src/test/java/org/apache/sysds/test/functions/builtin/BuiltinXgBoostTest_regression.java

```diff
@@ -0,0 +1,146 @@
+package org.apache.sysds.test.functions.builtin;
+
+import org.apache.sysds.common.Types;
+import org.apache.sysds.runtime.matrix.data.MatrixValue;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+
+public class BuiltinXgBoostTest_regression extends AutomatedTestBase {
+    private final static String TEST_NAME = "xgboost_regression";
+    private final static String TEST_DIR = "functions/builtin/";
+    private static final String TEST_CLASS_DIR = TEST_DIR + BuiltinXgBoostTest_regression.class.getSimpleName() + "/";
+    double eps = 1e-10;
+
+    @Override
+    public void setUp() {
+        TestUtils.clearAssertionInformation();
+        addTestConfiguration(TEST_NAME, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME, new String[]{"C"}));
+    }
+
+    @Parameterized.Parameter()
+    public int rows;
+    @Parameterized.Parameter(1)
+    public int cols;
+    @Parameterized.Parameter(2)
+    public int sml_type;
+    @Parameterized.Parameter(3)
+    public int num_trees;
+    @Parameterized.Parameter(4)
+    public double learning_rate;
+    @Parameterized.Parameter(5)
+    public int max_depth;
+    @Parameterized.Parameter(6)
+    public double lambda;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                {8, 2, 1, 2, 0.3, 6, 1.0},
```

Comment:
same argument here for parameterized testing

Reply:
done in 524629b0a6ce16c9a767ccbe6bac8042087070fa